news/tests.py

```diff
@@ -214,6 +221,47 @@ def toggle(pk, attr):
         self.assertEquals(toggle(self.article.pk, 'hidden'), {'color': 'grey' if hidden else 'yellow'})
         self.assertEquals(toggle(self.article.pk, 'hidden'), {'color': 'yellow' if hidden else 'grey'})
 
+    def test_event_ticket_emails_only_returns_active_tickeholders(self):
+        username_and_ticket_state_tuples = [
+            ("user1", False),
+            ("user1", True),
```

Comment:
The tests look much better now! I'm unsure, however, as to why you chose to write `"user1"` twice in this test (and the one below)..? This will create two tickets for the same user (which should be disallowed), and now when I read the value of `expected_string`, I can't really know which ticket the `user1@example.com` email comes from... 🤔 I'd suggest removing the first tuple in this list 🙂

Reply:
It should probably be two separate tests. The scenario here is that a person cancels their original ticket, and then signs up again.

`get_or_create` was used because the user does not exist, except for `user1` the second time. So it was used to catch both cases. If I swap it out with `get` I get the error `User matching query does not exist`.

Reply:
Oh, sorry, I meant `User.objects.create_user()` 😅

Reply:
I have added in the special case as a separate test, so I am keeping the `get_or_create`. As you can see from @sigtheidiot's [screenshot](https://github.com/MAKENTNU/web/pull/326#issuecomment-723480904). It is entirely possible to have two tickets for the same user.

Reply:
Cancelling a ticket is done by setting an existing ticket's `active` field to `False`, but the test code you've written simply creates two `EventTicket`s - one active and one inactive. It would absolutely be valuable to write a test for the `CancelTicketView`, though 😊 I don't see any value in writing a test that creates a ticket, sets the ticket's `active` field to `False`, and then checks the response, if that's what you're getting at, as that would pretty much be the same as just writing `False` in one of the tuples of `username_and_ticket_state_tuples` 🤔

> As you can see from @sigtheidiot's screenshot. It is entirely possible to have two tickets for the same user.

Yes, I know that it's possible, though it *shouldn't* be 😛 This isn't relevant to this PR, however.

Reply:
So you should not be able to reregister an event? Or are you saying that if the person reregisters, they should keep their original ticket, where `active` is set to `True` again?

Reply:
I just noticed that I hadn't read the existing code as thoroughly as I thought, so the tests you wrote makes sense now 🙂

> Or are you saying that if the person reregisters, they should keep their original ticket, where active is set to True again?

Yes, that sounds the most sane to me, as I don't really see any value in keeping a "log" of the times a user cancels and re-registers for an event 🤔